tests/_autospec.py

```diff
@@ -0,0 +1,64 @@
+import contextlib
+import functools
+import unittest.mock
+from typing import Callable
+
+
+@functools.wraps(unittest.mock._patch.decoration_helper)
+@contextlib.contextmanager
+def _decoration_helper(self, patched, args, keywargs):
+    """Skips adding patchings as args if their `dont_pass` attribute is True."""
+    # Don't ask what this does. It's just a copy from stdlib, but with the dont_pass check added.
+    extra_args = []
+    with contextlib.ExitStack() as exit_stack:
+        for patching in patched.patchings:
+            arg = exit_stack.enter_context(patching)
+            if not getattr(patching, "dont_pass", False):
+                # Only add the patching as an arg if dont_pass is False.
+                if patching.attribute_name is not None:
+                    keywargs.update(arg)
+                elif patching.new is unittest.mock.DEFAULT:
+                    extra_args.append(arg)
+
+        args += tuple(extra_args)
+        yield args, keywargs
```

Comment:
Since this is mostly copied, whatever license its under should be probably followed. [This license](https://github.com/python/cpython/blob/master/Lib/unittest/__init__.py#L29-L44) is for `unittest`. The mock module does not have its own license, only a comment that says its maintained by Michael Foord. Maybe the [PSF license](https://docs.python.org/3/license.html#psf-license-agreement-for-python-release) should be followed. I don't know, this is not straight forward.